### DISCUSSION TITLE
azure-pipeline-iac-flow-decision

### DISCUSSION THREAD

**Discussion by giscus** [WorkSafeBC-Common-Engineering/Common.Engineering.Discussion#19] [Technical Blog]
# azure-pipeline-iac-flow-decision

https://wsbctechnicalblog.github.io/azure-pipeline-iac-flow-decision.html

**Comment by AndreasNVI** [comment]
My thoughts on this:
Suppose we already have the application in production, but now we are adding a feature which requires that the infrastructure configuration be changed.
If we are using the Aggregated Flow with the adjusted configuration, we are able to update the infrastructure and then deploy the updated application. At this point we can test this out - both the infrastructure changes and the update application within a single environment before deploying to the next. The higher environments are not affected and will continue to run with the older configuration and application.

However if we go with Separated Flow, the updated infrastructure is deployed to all environments first, and then the application is deployed. If the new configuration is incorrect, it would then be broken in all the environments. Likewise if new configuration is correct but the older version of the application will break with it, then once all the infrastructure changes are in place all of the applications (still at the older version) are broken and will not run. This includes the exiting Production version of the application.

Being able to test the infrastructure changes with the updated application in just a single stage just seems to make more sense. The later environments are not affected and will continue to run.

**Reply by wpschaub** [reply]
@AndreasNVI another pro for the aggregated flow option - you highlighted a really good point of being able to validate the provisioning of the infrastructure and the deployment of the solution in development and then move on to subsequent stages.
